app/src/main/java/com/everest/mydetails/Validator.kt

```diff
@@ -0,0 +1,59 @@
+package com.everest.mydetails
+
+import android.text.TextUtils
+import android.widget.EditText
+
+class Validator {
+    var homeActivity: HomeActivity = HomeActivity()
+
+    fun isInputValid(
+        firstName: EditText,
+        age: EditText,
+        bio: EditText,
+        phoneNumber: EditText
+    ): Boolean {
+        if (isDataFilled(
+                firstName,
+                age,
+                bio,
+                phoneNumber
+            ) && isAgeValid(age) && isPhoneNumberValid(phoneNumber)
+        ) {
+            return true
+        }
+        return false
+    }
+
+    private fun isDataFilled(
+        firstName: EditText,
+        age: EditText,
+        bio: EditText,
+        phoneNumber: EditText
+    ): Boolean {
+        if (TextUtils.isEmpty(firstName.text) or TextUtils.isEmpty(age.text) or TextUtils.isEmpty(
+                bio.text
+            ) or TextUtils.isEmpty(phoneNumber.text)
+        ) {
+            homeActivity.getToastMessage(homeActivity.getString(R.string.fields_required_toast_message))
+            return false
+        }
+        return true
+    }
+
+    private fun isAgeValid(age: EditText): Boolean {
+        if (Integer.parseInt(age.text.toString()) > 120) {
```

Comment:
Since we passed edit text here , we do not have a generic age validation method
Should it just take a age as Int and then validate it?